### ISSUE TITLE
Incorrect candle messages on v3 websocket

### ISSUE THREAD

**Issue by example6** [Bittrex/bittrex.github.io#259]
I'm using the v3 WebSocket (https://socket-v3.bittrex.com/signalr) and C# example code to connect to the 5-min 'candle' channel for each currency pair available in the US. As messages roll forward into the next 5-minute interval, I cache the old candle values. 

For my own sanity, I have a test that checks the past few hours in the cache against the /v3/markets/.../recent REST API. Over the past few days, I've begun to see occurrences of candle messages that do not match what's found on the REST API.

For instance, here's a log of updates received for CELO-USD, between 1620023100 and 1620023700. The format of each message displayed is `[currency pair]-[timestamp]-[sequence number] - o:[open] c:[close] h:[high] l:[low]`
```
2021-05-02 23:25:02.925 -07:00 [VRB] CELO-USD-1620023100-10774 - o:5.5685 c:5.5685 h:5.5685 l:5.5685 v:0 q:0
2021-05-02 23:25:03.967 -07:00 [VRB] CELO-USD-1620023100-876 - o:5.564 c:5.564 h:5.564 l:5.564 v:0 q:0
2021-05-02 23:26:01.884 -07:00 [VRB] CELO-USD-1620023100-877 - o:5.574 c:5.574 h:5.574 l:5.574 v:44.85109436 q:249.99999996
2021-05-02 23:28:01.236 -07:00 [VRB] CELO-USD-1620023100-878 - o:5.574 c:5.567 h:5.574 l:5.567 v:76.587391 q:426.67596335
2021-05-02 23:30:01.942 -07:00 [VRB] CELO-USD-1620023400-879 - o:5.571 c:5.571 h:5.571 l:5.571 v:31.56881244 q:175.8698541
2021-05-02 23:30:02.975 -07:00 [VRB] CELO-USD-1620023400-10816 - o:5.576 c:5.576 h:5.576 l:5.576 v:0 q:0
2021-05-02 23:30:03.633 -07:00 [VRB] CELO-USD-1620023400-880 - o:5.567 c:5.567 h:5.567 l:5.567 v:0 q:0
2021-05-02 23:35:03.160 -07:00 [VRB] CELO-USD-1620023700-881 - o:5.571 c:5.571 h:5.571 l:5.571 v:0 q:0
2021-05-02 23:35:03.576 -07:00 [VRB] CELO-USD-1620023700-10829 - o:5.5775 c:5.5775 h:5.5775 l:5.5775 v:0 q:0

```

The last update I got for CELO-USD in the window 1620023400 had an open of 5.567, but this doesn't match what's been recorded:
`{"startsAt":"2021-05-03T06:30:00Z","open":"5.57100000","high":"5.57100000","low":"5.57100000","close":"5.57100000","volume":"31.56881244","quoteVolume":"175.86985410"}`

The previous entry in the sequence (879) actually has the correct value. 

Each instance I've found (so far) of this occurring seems to follow the pattern of a message having an open different from the previous message's close, despite reporting zero volumes on the new message. 

The past few times I've seen this happen are: 
* GAME-BTC @ 1620017100
* RVN-BTC @ 1620013200
* BSV-ETH @ 1619981700
* BSV-USDT @ 1619981700

I couldn't find any documentation about this. Should I drop messages with changed value that have zero volume? I am not sure what other incorrect situations could arise from that, but I'd like to feel confident that I have a good source of truth.

**Comment by bittrex-pete** [comment]
Hi @example6 - thanks for the report. That's all very strange. Sorry that the documentation isn't more helpful with respect to what 0-volume candles are and what they mean. I've logged a task for myself to resolve that. Let me digress for a moment to explain what these things are. The socket makes a point of sending you a message for each candle stream at the start of each interval (e.g. every 5 minutes for each market). If there has not been any trading activity in the interval yet when the message is sent, it sends what is essentially a placeholder with 0 volume that carries over the Close value of the previous candle as OHLC. 

When/if trading activity occurs, an update is sent for that candle that contains real information and has >0 volume. We do this because it makes life a lot easier for applications that are showing live charts to their users. Depending on your use case, you could potentially ignore all 0-volume candles. That would just result in you not always having data for each candle interval. Otherwise the placeholders shouldn't require any special handling since they come once at the start of the interval and are then replaced by an update later if trading activity occurs.

However there are a few things I can see in your log that are unexpected. Between the first and second messages (10774 and 876) the OHLC values have changed, but there is zero volume reported. This is also true between a number of the other 0 volume messages. That doesn't make sense since those values shouldn't change without any trading activity. Another thing I see in your log that shouldn't happen is that a "real" candle from message 879 is being followed by a 0-volume placeholder in message 880.

Additionally, there are a variety of odd things happening with the sequence numbers. You show sequence number starting at 10774 and then dropping to 876. Sequence number resets do happen, but when they do they should go from a high number to a close to 0 number and we should have a record of it in our logs. I don't see any record of one happening in this time frame. Also the first message after a reset being numbered 876 would imply that you somehow didn't get >800 messages after a reset.  It's also really weird that the sequence numbers in your log seem to be bouncing back and forth between >10K and <900. 

To summarize, there's a bunch of stuff in your log I wouldn't expect to see.  Can you confirm a few things?
1. This log data is exclusively from the CELO-USD 5 minute candle stream (i.e. no possibility of there being data from other streams mixed in; such as accidentally including every fifth interval from the 1 minute candle stream)
2. This includes all the messages received on the CELO-USD 5 minute candle stream for the logged period 
3. The sequence numbers included in your log are the ones you received from the system

**Comment by example6** [comment]
Hi @bittrex-pete, thank you for the response and detailed explanation of zero-volume candles. 

To confirm your questions:
1. Yes, my application only subscribes to 5-minute candle streams, and these log entries are grepped from my main log. I've checked over my code again and don't see anything that might cause those values to be displayed with the wrong sequence number.
2. Yes, these are all the messages received between the two timestamps listed above. These log entries come before any logic happens to drop messages with invalid sequence (hence the >10K seq. numbers). I will attach the full log for CELO-USD during this time period: [CELO-USD-support.log](https://github.com/Bittrex/bittrex.github.io/files/6417993/CELO-USD-support.log)
3. Yes, these are all sequence numbers received on the candle channel

A few extra details:
* I changed my code to drop messages where `next.Open != prev.Close && next.volume ==0`, and over the past ~14 hours it seems like I've been able to get a consistent, complete, correct stream of data, so at least it seems like there's a good workaround. I will continue to look over the data and see if any other problem cases arise.
* I added some output to this case, and found that it happens usually at least once per 5-min interval, on at least one market. In most cases, I do get a "correct" message before the next interval, which seems to hide the issue.
* I have observed those unexpected sequence numbers on every market since I began working with this websocket API a few weeks ago, and I had concluded that it was an intentionally tricky obstacle to force users to actually use the market snapshot for each currency pair. If that's not the case, I look forward to being able to simplify my code and reduce my API usage. 

Please let me know if there is any other data I can provide, or if your team needs any help to reproduce these issues.

**Comment by bittrex-pete** [comment]
Thanks for the details. It looks like there is a problem on our side that's causing this. We'll get it fixed. Thanks again for reporting it!

**Comment by bittrex-pete** [comment]
Hi @example6 - are you still seeing this behavior? I believe it was fixed last week.

**Comment by example6** [comment]
Hi @bittrex-pete, sorry for the late response. Unfortunately it looks like I am still seeing this. Here are some examples:

```
2021-05-28 06:25:00.818 -07:00 [VRB] BTC-USDT-1622208000-66106 - o:36873.49499996 c:36846.85635849 h:36888.82988794 l:36773.52856503 v:3.28264832 q:120922.98676554
2021-05-28 06:25:01.858 -07:00 [VRB] BTC-USDT-1622208000-66107 - o:36873.49499996 c:36849.83618496 h:36888.82988794 l:36773.52856503 v:3.28264832 q:120922.98676554
2021-05-28 06:25:02.735 -07:00 [VRB] BTC-USDT-1622208300-66108 - o:36849.83618496 c:36849.83618496 h:36849.83618496 l:36849.83618496 v:0 q:0
2021-05-28 06:25:04.127 -07:00 [VRB] BTC-USDT-1622208300-66109 - o:36858.44236501 c:36858.44236501 h:36858.44236501 l:36858.44236501 v:0 q:0
2021-05-28 06:25:05.216 -07:00 [VRB] BTC-USDT-1622208300-66110 - o:36858.44236501 c:36850.94670247 h:36858.44236501 l:36850.94670247 v:0 q:0
2021-05-28 06:25:06.243 -07:00 [VRB] BTC-USDT-1622208300-66111 - o:36858.44236501 c:36850.94670247 h:36858.44236501 l:36850.94670247 v:0.01200238 q:442.29906568
2021-05-28 06:25:08.258 -07:00 [VRB] BTC-USDT-1622208300-66112 - o:36858.44236501 c:36850.94670246 h:36858.44236501 l:36850.94670246 v:0.01200238 q:442.29906568
2021-05-28 06:25:09.201 -07:00 [VRB] BTC-USDT-1622208300-66113 - o:36858.44236501 c:36845.47335123 h:36858.44236501 l:36845.47335123 v:0.01200238 q:442.29906568
```

According to your specification, I believe the update `BTC-USDT-1622208300-66108` is valid with zero-volume, since it's the beginning of a new interval, but the two updates just after that seem to change values in the same interval without reporting changed volume. 

Here's the same thing happening with ETH-USDT:
```
2021-05-28 04:19:27.478 -07:00 [VRB] ETH-USDT-1622200500-5187 - o:2500 c:2509.66305051 h:2516.39444251 l:2500 v:6.79811733 q:17030.29061675
2021-05-28 04:19:45.715 -07:00 [VRB] ETH-USDT-1622200500-5188 - o:2500 c:2511.32163171 h:2516.39444251 l:2500 v:10.30746733 q:25843.39718498
2021-05-28 04:20:01.903 -07:00 [VRB] ETH-USDT-1622200800-5189 - o:2510 c:2510 h:2510 l:2510 v:1 q:2510
2021-05-28 04:20:03.918 -07:00 [VRB] ETH-USDT-1622200800-5190 - o:2511.32163171 c:2511.32163171 h:2511.32163171 l:2511.32163171 v:0 q:0
2021-05-28 04:20:13.536 -07:00 [VRB] ETH-USDT-1622200800-5191 - o:2510 c:2507.34860224 h:2510 l:2507.34860224 v:2.82745186 q:7092.05886683
2021-05-28 04:20:59.596 -07:00 [VRB] ETH-USDT-1622200800-5192 - o:2510 c:2497.44279933 h:2510 l:2497.44279933 v:2.90738988 q:7291.69949927
2021-05-28 04:21:25.001 -07:00 [VRB] ETH-USDT-1622200800-5193 - o:2510 c:2492.286 h:2510 l:2492.286 v:2.91024748 q:7298.82145574
2021-05-28 04:21:27.108 -07:00 [VRB] ETH-USDT-1622200800-5194 - o:2510 c:2491.892 h:2510 l:2491.892 v:3.03654355 q:7613.5376222
```

Here, `ETH-USDT-1622200800-5189` has a good value, with volume, at the start of the candle interval, and the one immediately after that, in the same interval, has changed values without changed volume.

As far as I can tell, each of these are now happening right after a new interval, so the behavior does seem to have changed. (Previously it was independent of interval start time) Looking at my logs, it seems like about one instance of this is happening on a different market every 10-15 minutes.

Please let me know if there's any additional information I can provide.

**Comment by bittrex-pete** [comment]
Thanks for the detailed follow-up @example6! I've asked someone to investigate.
